### PR TITLE
Add CI for testing, linting, and deployment

### PR DESCRIPTION
**Pull request by velovix**
Adds two new Github Actions workflows to this repository:

Run tests: This workflow runs tests with pytest and lints the code. Note that the linter doesn't result in a build failure right now. It just logs warnings for the time being. This workflow runs on PRs headed to master only.

Publish packages: Packages up the vcap and vcap_utils packages and uploads them to PyPI. These packages now use setuptools_scm to figure out their version. If the currently deployed commit has a version tag, it will be that version. Otherwise, it will be `major.minor.patch+1.dev{commits since last tag}`.

### CODE REVIEW

**Comment by velovix** [comment]
I know that this PR has a lot of commits, though we squash and merge on this project so it will all be one commit on master.

File: vcap/setup.py [hunk 1]
```diff
@@ -5,11 +5,24 @@
...
 setup(
     name='vcap',
-    version='0.1.3',
     description="A library for creating OpenVisionCapsules in Python",
+    author="Dilili Labs",
     packages=find_namespace_packages(include=["vcap*"]),
 
-    author="Dilili Labs",
+    # Pull the package version from Git tags
+    use_scm_version={
```
Comment (apockill) [inline vcap/setup.py:13]: Does this require the user to have setuptools_scm pre-installed, or is this something that is supported by pip/setuptools somehow?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.6']
+    
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        # Include tags in the checkout
+        fetch-depth: 0
```
Comment (apockill) [inline .github/workflows/deploy.yml:18]: Ha, that's awesome

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        key: ${{ runner.os }}-${{ hashFiles('**/setup.py') }}-pip-cache
+        restore-keys: |
+          ${{ runner.os }}-pip-cache
+    - name: Install dependencies
+      run: |
+        pip install --upgrade setuptools wheel twine
+    - name: Publish vcap
+      env:
+        TWINE_USERNAME: __token__
+        TWINE_PASSWORD: ${{ secrets.TEST_PYPI_PASSWORD }}
```
Comment (apockill) [inline .github/workflows/deploy.yml:40]: To confirm- when we do ordinary deployments we won't be able to use this. This pipeline only deploys to the pypi testserver.

Comment (velovix) [reply .github/workflows/deploy.yml:40]: Oops!

File: vcap/setup.py (hunk 1 above)
Comment (velovix) [reply vcap/setup.py:13]: I didn't have to pip install setuptools_scm myself, so I think it's managed by setuptools. I just tried it in a fresh virtualenv and it worked fine. Not sure exactly how this works internally though.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (velovix) [reply .github/workflows/deploy.yml:18]: Glad we don't have any big files in our history!

File: vcap/setup.py (hunk 1 above)
Comment (apockill) [reply vcap/setup.py:13]: Awesome!

**Pull request merged**